src/syslog-counter/src/lib.rs

```diff
@@ -0,0 +1,134 @@
+use std::io::{self, BufRead, Write};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+use log::{error, info};
+use std::process::exit;
+use swss_common::*;
+
+pub fn convert_to_CxxString(opt: Option<u64>) -> Option<CxxString> {
+    opt.map(|num| CxxString::new(num.to_string()))
+}
+
+pub fn convert_to_u64(opt: Option<&CxxString>) -> Option<u64> {
+    opt.and_then(|s| match s.to_str() {
+        Ok(text) => text.parse::<u64>().ok(),
+        Err(_) => None,
+    })
+}
```

Comment:
To me, it doesn't make sense for these conversion functions to take in `Option<T>`, considering for these functions to do any work, that argument is required. I'd recommend having them take in the actual value, and then either return the actual target type, or `Result<T, E>` if the conversion could fail.

@r12f for his thoughts on this.